types/types.cc

```diff
@@ -350,7 +350,14 @@ static uint32_t serialize(sstring_view input, int64_t days) {
 }
 uint32_t simple_date_type_impl::from_sstring(sstring_view s) {
     char* end;
+    errno = 0;
     auto v = std::strtoll(s.begin(), &end, 10);
+     if (errno == ERANGE) {
+        throw marshal_exception(format("Unable to make unsigned int (for date): the value entered is too large (more than 64 bits)"));
+    }
+    if (v < 0 || v > std::numeric_limits<uint32_t>::max()) {
+        throw marshal_exception(format("Unable to make unsigned int (for date): the value entered overflows unsigned int32"));
+    }
```

Comment:
As I noted above, I think the word "overflows" is confusing for a negative number. I'm also not sure we need the extra information about 64 bits. I think the message should just say that it needs to fit an unsigned 32-bit integer, which is exactly what the first part of the message says. I would just use the same message as Cassandra.